whois-api/src/test/java/net/ripe/db/whois/api/rdap/RdapServiceTestIntegration.java

```diff
@@ -3127,6 +3128,422 @@ public void get_help_response() {
+
+        final SearchResult searchResult = createResource("ips/rirSearch1/down/192.0.2.0/24")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(SearchResult.class);
+
+        final List<Ip> ipResults = searchResult.getIpSearchResults();
+        assertThat(ipResults.size(), is(2));
+        assertThat(ipResults.getFirst().getHandle(), is("192.0.2.0 - 192.0.2.127"));
+        assertThat(ipResults.get(1).getHandle(), is("192.0.2.128 - 192.0.2.255"));
+    }
+
+    @Test
+    public void get_down_ipv6_then_immediate_child(){
+        loadIpv6RelationTreeExample();
+
+        final SearchResult searchResult = createResource("ips/rirSearch1/down/2000::/3")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(SearchResult.class);
+
+        final List<Ip> ipResults = searchResult.getIpSearchResults();
+        assertThat(ipResults.size(), is(4));
+        assertThat(ipResults.getFirst().getHandle(), is("2001::/16"));
+        assertThat(ipResults.get(1).getHandle(), is("2400::/12"));
+        assertThat(ipResults.get(2).getHandle(), is("2600::/12"));
+        assertThat(ipResults.get(3).getHandle(), is("2800::/12"));
+    }
+
+    @Test
+    public void get_down_domain_then_immediate_child(){
+        loadIpv4RelationTreeExample();
+        loadIpv4RelationDomainExample();
+
+        final SearchResult searchResult = createResource("domains/rirSearch1/down/0.192.in-addr.arpa")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(SearchResult.class);
+
+        final List<Domain> domainResults = searchResult.getDomainSearchResults();
+        assertThat(domainResults.size(), is(1));
+        assertThat(domainResults.getFirst().getHandle(), is("2.0.192.in-addr.arpa"));
+    }
+
+    @Test
+    public void get_down_domain_no_child_then_empty_response(){
+        loadIpv4RelationTreeExample();
+        loadIpv4RelationDomainExample();
+
+        final SearchResult searchResult = createResource("domains/rirSearch1/down/1.2.0.192.in-addr.arpa")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(SearchResult.class);
+
+        final List<Domain> domainResults = searchResult.getDomainSearchResults();
+        assertThat(domainResults, is(nullValue()));
+    }
+
+    @Test
+    public void get_down_when_no_child_then_empty_response(){
+        loadIpv4RelationTreeExample();
+
+        final SearchResult searchResult = createResource("ips/rirSearch1/down/192.0.2.0/32")
+                .request(MediaType.APPLICATION_JSON_TYPE)
+                .get(SearchResult.class);
+
+        final List<Ip> ipResults = searchResult.getIpSearchResults();
+        assertThat(ipResults, is(nullValue()));
+    }
+
+    @Test
+    public void down_with_status_then_501(){
+
+        final ServerErrorException notImplementedException = assertThrows(ServerErrorException.class, () -> {
+            createResource("ip/rirSearch1/down/192.0.2.0/24?status=inactive")
```

Comment:
this should be returning error for the type (invalid type)
use ips in the example